lib/engine.js

```diff
@@ -17,80 +17,96 @@
 \*───────────────────────────────────────────────────────────────────────────*/
 'use strict';
 
-var path = require('path'),
-    dust = require('dustjs-linkedin'),
-    utils = require('./utils'),
-    patch = require('./patch'),
-    River = require('./brook'),
-    reqwire = require('./reqwire');
+var freshy = require('freshy').freshy;
+var path = require('path');
+var utils = require('./utils');
+var River = require('./brook');
+var reqwire = require('./reqwire');
+var dustjacket = require('dustjacket');
+var fs = require('fs');
 
 
-var loaders = {
-    dust: require('./reader/dust'),
-    js: require('./reader/js')
-};
+function loader (name, context, cb) {
+    var views = utils.resolveViewDir(context.global);
 
+    var ext = path.extname(name) || 'dust';
+    if (ext[0] !== '.') {
+        ext = '.' + ext;
+    }
 
+    var file = path.resolve(views, name + ext);
+    fs.readFile(file, 'utf8', function (err, content) {
+        if (err) {
+            if (err.code !== 'ENOENT') {
+                cb(err);
+            } else {
+                cb();
+            }
+        } else {
+            cb(null, content );
+        }
+    });
+}
 
-function createRenderContext(options) {
-    var context;
+exports.create = function (config) {
+    config = config || {};
 
-    // Hoist *only* view, settings, and ext to global context.
-    context = dust.makeBase({
-        views: options.views,
-        settings: options.settings,
-        ext: options.ext
-    });
+    var dust = freshy('dustjs-linkedin', function (dust) {
+        // In this context, the currently-cached dustjs-linkedin is the one we're requiring,
+        // but that will be expunged after this function ends, so we load this now so 
+        // it's always available. All this is synchronous.
+        require('dustjs-helpers');
+
+        // Load referenced helpers
+        if (Array.isArray(config.helpers)) {
+            config.helpers.forEach(function (module) {
+                // allow configuration of helpers like:
+                // "helpers": ["./lib/abc", { "name": "./lib/test", "arguments": { "debug": true } }] 
+                if (typeof module === 'string') {
+                    reqwire.init(module, dust);
+                }
+                else if (module.hasOwnProperty('name')) {
+                    reqwire.init(module.name, dust, module.arguments);
+                }
+            });
+        }
 
-    // Remove from child context so global values are used.
-    delete options.views;
-    delete options.settings;
-    delete options.ext;
+        dustjacket.registerWith(dust);
 
-    context = context.push(options);
-    return context;
-}
+        if (config.loaders) {
+            config.loaders.forEach(function (loader) {
+                dust.addLoadMiddleware(loader);
+            });
```

Comment:
[NITPICK 1]
Will probably want the same check as above wrt `config.helpers`, here:
```javascript
if (Array.isArray(config.loaders)) {
```
[NITPICK 2]
Could be simplified to:
```javascript
config.loaders.forEach(dust.addLoadMiddleware);
```
[NITPICK 3]
Might be worth noting in comments that `addLoadMiddleware` is added via `dustjacket` decoration and not standard with dust. This will hopefully guide maintainers in the right direction should someone try to debug this code.